### PR TITLE
Implement on the Snapshot Testing with report + PR comment by Github Action

### PR DESCRIPTION
**Pull request by kittinunf**
### What's in this PR?

This PR is probably an extremely poor-man, quick-and-dirty solution to achieve what we might be looking for kub.

1. We run `verifyRoborazziDebug` by the CI
2. Check whether compare files are found
  - If yes, generate a report and comment it back into PR
  - If no, do nothing
3. In any case, we upload artifacts into our checks

image

In this PR, I also add a phantom type `SnapshotTests` to categorize test classes so that the snapshot test is only run by the gradle task with (-Psnapshot) parameter in the `verifyRoborazzi` call.

### CODE REVIEW

**Comment by kittinunf** [comment]
It looks like `GITHUB_TOKEN` doesn't have enough permission kub. I think someone need to add correct GITHUB_TOKEN (with permission to add comment into the PR for me) 🙏

**Comment by akexorcist** [comment]
ผมรู้สึกว่ามันถูกแล้วนะ เกี่ยวกับที่ว่าเป็น pr จากพี่หรือป่าวนะ 🤔

**Comment by akexorcist** [comment]
"Note: In public repositories this action does not work in pull_request workflows when triggered by forks. Any attempt will be met with the error, Resource not accessible by integration. This is due to token restrictions put in place by GitHub Actions. Private repositories can be configured to ..."

ไปเจอมาจาก https://github.com/peter-evans/create-or-update-comment?tab=readme-ov-file#action-inputs เดี๋ยวผมขอลองเช็คให้ก่อนนะครับ

File: .github/workflows/snapshot_test.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+name: SnapshotTest
+
+on:
+  workflow_dispatch:
+  pull_request:
```
Comment (akexorcist) [inline .github/workflows/snapshot_test.yml:5]: Replace `pull_request` with `pull_request_target`?

Comment (kittinunf) [reply .github/workflows/snapshot_test.yml]: 👍 💯

File: app/src/main/java/com/akexorcist/ruammij/ui/overview/OverviewScreen.kt [hunk 2]
```diff
@@ -119,6 +120,7 @@ private fun OverviewScreen(
 ) {
     Column(
         modifier = Modifier
+            .background(color = Color.Red)
```
Comment (kittinunf) [inline app/src/main/java/com/akexorcist/ruammij/ui/overview/OverviewScreen.kt:123]: This is to be removed before merge kub

**Comment by akexorcist** [comment]
It's worked!

**Comment by akexorcist** [comment]
No `SnapshotTest` anymore 😂

image

**Comment by kittinunf** [comment]
Lol ... Shall we revert back to `pull_request`?

**Comment by akexorcist** [comment]
Let me try to reopen this pull request first krub

**Comment by kittinunf** [comment]
Nope ... negative lol...

**Comment by kittinunf** [comment]
Nope haha ... but I think you are right kub, it might be something to do with my fork repo :)

**Comment by akexorcist** [comment]
Another option is invite you into this project but we shouldn't do that 😂

**Comment by kittinunf** [comment]
If you can't invite then, it is okay kub, you can take my work from here and continue further

**Comment by akexorcist** [comment]
Yes, I can. But it doesn't make sense if we have to invite every contributors into the project 😂

So the better approach is do something that this thing works!

**Comment by akexorcist** [comment]
I think you should push your branch into origin repository then recreate the pull request to make the Actions works

**Comment by kittinunf** [comment]
Thanks kub! Accepted! Wait ... the report seems to be working but we don't have a comment 🤔 ?

**Comment by akexorcist** [comment]
@kittinunf it's the result from #53 

Try to create your own PAT then put it in secrets with 'SNAPSHOT_TEST_RESULT_COMMENT' key name.

**Comment by kittinunf** [comment]
Ok!! Let me re-create this PR without the fork then :)

**Comment by kittinunf** [comment]
Supersede with #56

**Pull request closed without merging**